components/libc/libdl/dlelf.c

```diff
@@ -473,3 +475,457 @@ rt_err_t dlmodule_load_relocated_object(struct rt_dlmodule* module, void *module
 
     return RT_EOK;
 }
+
+#elif (defined(__aarch64__) || defined(__x86_64__) || (__riscv_xlen == 64)) 
+rt_err_t dlmodule_load_shared_object(struct rt_dlmodule* module, void *module_ptr)
+{
+    rt_bool_t linked   = RT_FALSE;
+    rt_uint32_t index, module_size = 0;
+    Elf32_Addr vstart_addr, vend_addr;
+    rt_bool_t has_vstart;
+
+    RT_ASSERT(module_ptr != RT_NULL);
+
+    if (rt_memcmp(elf_module->e_ident, RTMMAG, SELFMAG) == 0)
+    {
+        /* rtmlinker finished */
+        linked = RT_TRUE;
+    }
+
+    /* get the ELF image size */
+    has_vstart = RT_FALSE;
+    vstart_addr = vend_addr = RT_NULL;
+    for (index = 0; index < elf_module->e_phnum; index++)
+    {
+        if (phdr[index].p_type != PT_LOAD)
+            continue;
+
+        //LOG_D("LOAD segment: %d, 0x%p, 0x%08x", index, phdr[index].p_vaddr, phdr[index].p_memsz);
+        if (phdr[index].p_memsz < phdr[index].p_filesz)
+        {
+            rt_kprintf("invalid elf: segment %d: p_memsz: %d, p_filesz: %d\n",
```

Comment:
Please use LOG_E to replace rt_kprintf.

Reply:
OK,it's a historical issue and I wiil fix it